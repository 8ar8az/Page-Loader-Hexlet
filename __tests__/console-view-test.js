import consoleView from '../src/view';

const standardConsoleLog = console.log;
const standardConsoleError = console.error;
const consoleLogOutput = [];
const consoleErrorOutput = [];

const consoleLogMock = (arg) => {
  consoleLogOutput.push(arg);
};
const consoleErrorMock = (arg) => {
  consoleErrorOutput.push(arg);
};

beforeEach(() => {
  console.log = consoleLogMock;
  console.error = consoleErrorMock;
});

afterEach(() => {
  console.log = standardConsoleLog;
  console.error = standardConsoleError;
});

test("Show statuses of resource's saving", () => {
  const resources = [
    {
      resourceUrl: 'http://www.test.ru/assets/style.css',
      isSaved: true,
    },
    {
      resourceUrl: 'http://www.test.ru/assets/script.js',
      isSaved: true,
    },
    {
      resourceUrl: 'http://www.test.ru/assets/img.jpg',
      isSaved: false,
      err: new Error('Something wrong'),
    },
  ];

  consoleView.showStatusOfResourcesSaving(resources);

  expect(consoleLogOutput).toMatchSnapshot();
  expect(consoleErrorOutput).toMatchSnapshot();
});
