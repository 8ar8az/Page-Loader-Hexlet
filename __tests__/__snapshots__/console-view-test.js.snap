// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show statuses of resource's saving 1`] = `
Array [
  "This local resources has been successful saved:",
  "+ http://www.test.ru/assets/style.css",
  "+ http://www.test.ru/assets/script.js",
  undefined,
]
`;

exports[`Show statuses of resource's saving 2`] = `
Array [
  "This local resources hasn't been saved:",
  "- http://www.test.ru/assets/img.jpg",
  "Reason: Something wrong",
]
`;
